### ISSUE TITLE
Two interactions that only differ by query string

### ISSUE THREAD

**Issue by tshortt** [pact-foundation/pact-js#12]
Is it possible to add two different interactions that only differ by query string. Here's an example of what I'm trying to accomplish:

`projectsProvider.addInteraction({
        uponReceiving: 'a request for employee',
        withRequest: {
            method: 'get',
            path: '/EmployeeApi/api/organizations/0/employees'
        }
`

versus

`projectsProvider.addInteraction({
        uponReceiving: 'a request for employee with emails',
        withRequest: {
            method: 'get',
            path: '/EmployeeApi/api/organizations/0/employees',
            query: {
                'include': 'emails'
            }
        }
`

Error:
Failed: Actual interactions do not match expected interactions for mock MockService.

Missing requests:
        GET /EmployeeApi/api/organizations/0/employees?include=emails

**Comment by mefellows** [comment]
Hi @tshortt, yes that should be possible - uniqueness for a request is calculated as a combination of the attributes in the `withRequest` block.

Could you please provide a Gist to some code? It's more than likely a setup problem.

**Comment by tarciosaraiva** [comment]
Following up on @mefellows answer above, @tshortt could you please also provide the log generated by your test? When the mock server runs it creates a log under `<project>/logs/mockserver-integration.log` that would contain the interactions setup / received.

**Comment by tarciosaraiva** [comment]
@tshortt I just realised I didn't have any test related to this case so I'm adding one now. Here's the output I got from my mock server log:

```
I, [2016-09-21T09:03:53.219929 #14094]  INFO -- : Registered expected interaction GET /projects?from=today
D, [2016-09-21T09:03:53.221251 #14094] DEBUG -- : {
  "description": "a request for projects",
  "provider_state": "i have a list of projects",
  "request": {
    "method": "GET",
    "path": "/projects",
    "query": {
      "from": [
        "today"
      ]
    },
    "headers": {
      "Accept": "application/json"
    }
  },
  "response": {
    "status": 200,
    "headers": {
      "Content-Type": "application/json"
    },
    "body": [
      {
        "id": 1,
        "name": "Project 1",
        "due": "2016-02-11T09:46:56.023Z",
        "tasks": [
          {
            "id": 1,
            "name": "Do the laundry",
            "done": true
          },
          {
            "id": 2,
            "name": "Do the dishes",
            "done": false
          },
          {
            "id": 3,
            "name": "Do the backyard",
            "done": false
          },
          {
            "id": 4,
            "name": "Do nothing",
            "done": false
          }
        ]
      }
    ]
  }
}
I, [2016-09-21T09:03:53.234556 #14094]  INFO -- : Received request GET /projects?from=today
D, [2016-09-21T09:03:53.235412 #14094] DEBUG -- : {
  "path": "/projects",
  "query": "from=today",
  "method": "get",
  "headers": {
    "Host": "localhost:9744",
    "Accept-Encoding": "gzip, deflate",
    "User-Agent": "node-superagent/2.2.0",
    "Accept": "application/json",
    "Connection": "close",
    "Https": "on",
    "Version": "HTTP/1.1"
  }
}
I, [2016-09-21T09:03:53.236346 #14094]  INFO -- : Found matching response for GET /projects?from=today
D, [2016-09-21T09:03:53.237597 #14094] DEBUG -- : {
  "status": 200,
  "headers": {
    "Content-Type": "application/json"
  },
  "body": [
    {
      "id": 1,
      "name": "Project 1",
      "due": "2016-02-11T09:46:56.023Z",
      "tasks": [
        {
          "id": 1,
          "name": "Do the laundry",
          "done": true
        },
        {
          "id": 2,
          "name": "Do the dishes",
          "done": false
        },
        {
          "id": 3,
          "name": "Do the backyard",
          "done": false
        },
        {
          "id": 4,
          "name": "Do nothing",
          "done": false
        }
      ]
    }
  ]
}
I, [2016-09-21T09:03:53.245853 #14094]  INFO -- : Verifying - interactions matched for example ""
I, [2016-09-21T09:03:53.255236 #14094]  INFO -- : Writing pact with details {:consumer=>{:name=>"Consumer 1"}, :provider=>{:name=>"Provider 1"}}
I, [2016-09-21T09:03:53.255425 #14094]  INFO -- : Writing pact for Provider 1 to /home/nab/workspace/pact-js/pacts/consumer_1-provider_1.json
I, [2016-09-21T09:03:53.264434 #14094]  INFO -- : Cleared interactions before example ""
```

Here's the test added: https://github.com/pact-foundation/pact-js/blob/master/test/dsl/integration.spec.js#L100

**Comment by tshortt** [comment]
Here's a gist
https://gist.github.com/tshortt/e453cf12ea7c3114f0c166dc9e75bdfa

So lines 34-36, versus lines 77-82

**Comment by tshortt** [comment]
@tarciosaraiva Trying to get my mockserver-integration.log in here in an acceptable format.

**Comment by tarciosaraiva** [comment]
hi @tshortt looks like it might be a bug with the pact mock server. I [opened a new issue there](https://github.com/bethesque/pact-mock_service/issues/54), let's see what comes back :+1:

**Comment by mefellows** [comment]
Marking as enhancement, waiting on bethesque/pact-mock_service/issues/54.

**Comment by mefellows** [comment]
I've just tested this and your scenario should work as expected. I'm not sure where along the lines this was fixed, but it is :)
